src/test/ui/parser-recovery-4.stderr

```diff
@@ -0,0 +1,34 @@
+error: incorrect close delimiter: `}`
+  --> $DIR/parser-recovery-4.rs:20:1
+   |
+LL | fn main() {
+   |           - close delimiter possibly meant for this
+LL |     let x = 1;
+LL |     foo(x
+   |        - un-closed delimiter
+...
+LL | }
+   | ^ incorrect close delimiter
+
+error: expected one of `!`, `)`, `,`, `.`, `::`, `?`, `{`, or an operator, found `bar`
+  --> $DIR/parser-recovery-4.rs:17:5
+   |
+LL |     foo(x
+   |        - -
+   |        | |
+   |        | expected one of 8 possible tokens here
+   |        | help: ...the missing `)` may belong here
+   |        if you meant to close this...
+LL |     bar()
+   |     ^^^ unexpected token
+
+error: expected one of `.`, `;`, `?`, `}`, or an operator, found `bar`
+  --> $DIR/parser-recovery-4.rs:17:5
+   |
+LL |     foo(x
+   |          - expected one of `.`, `;`, `?`, `}`, or an operator here
+LL |     bar()
+   |     ^^^ unexpected token
```

Comment:
In the ideal end state, this error would not get emitted, instead having the prior one be

```
error: expected one of `!`, `)`, `,`, `.`, `::`, `?`, `{`, or an operator, found `bar`
  --> $DIR/parser-recovery-4.rs:17:5
   |
LL |     foo(x
   |        - ^- help: you might also be missing a semicolon here: `;`
   |        | |
   |        | expected one of 8 possible tokens here
   |        | help: ...the missing `)` may belong here
   |        if you meant to close this...
LL |     bar()
   |     ^^^ unexpected token
```

but I'm not sure how to verify that we don't start suggesting bogus code that wouldn't work...

Reply:
I'm not sure I understand. But I do think that this example you just gave is kind of cluttered -- I feel like it'd be better to break it apart into multiple sections or something.